TypeCobol.Analysis/Cfg/ControlFlowGraphBuilder.cs

```diff
@@ -0,0 +1,2695 @@
+        /// </summary>
+        public enum CfgMode
+        {
+            Normal,
+            Extended,
+        }
+
+        /// <summary>
+        /// Set the CFG Mode
+        /// </summary>
+        public CfgMode Mode
+        {
+            get;
+            set;
+        }
+
+        /// <summary>
+        /// The parent program Control Flow Builder, for nested Program.
+        /// </summary>
+        public ControlFlowGraphBuilder<D> ParentProgramCfgBuilder
+        {
+            get;
+            private set;
+        }
+
+
+        /// <summary>
+        /// All Cfg graphs builder created during the building phase, so it contains Cfg for nested programs and nested procedures,
+        /// but also for stacked programs.
+        /// </summary>
+        public List<ControlFlowGraphBuilder<D>> AllCfgBuilder
+        {
+            get;
+            internal set;
+        }
+
+        /// <summary>
+        /// The current Program Cfg being built.
+        /// </summary>
+        public ControlFlowGraphBuilder<D> CurrentProgramCfgBuilder
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// The Control Flow Graph Build for the main Program
+        /// </summary>
+        public ControlFlowGraph<Node, D> Cfg
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// The Current Program symbol being built as a Scope
+        /// </summary>
+        public Program Program
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// The Current Program symbol being built as a Scope
+        /// </summary>
+        private Program CurrentProgram
+        {
+            get;
+            set;
+        }
+
+        /// <summary>
+        /// The current entered node.
+        /// </summary>
+        private Node CurrentNode
+        {
+            get;
+            set;
+        }
+
+        /// <summary>
+        /// The current basic block
+        /// </summary>
+        protected BasicBlockForNode CurrentBasicBlock
+        {
+            get;
+            set;
+        }
+
+        /// <summary>
+        /// The current section symbol.
+        /// </summary>
+        internal CfgSectionSymbol CurrentSection
+        {
+            get;
+            set;
+        }
+
+        /// <summary>
+        /// The current paragraph symbol.
+        /// </summary>
+        internal CfgParagraphSymbol CurrentParagraph
+        {
+            get;
+            set;
+        }
+
+        /// <summary>
+        /// The current Sentence in the current Paragraph.
+        /// </summary>
+        internal CfgSentence CurrentSentence
+        {
+            get;
+            set;
+        }
+
+        /// <summary>
+        /// The Section and Paragraph Domain of this program.
+        /// </summary>
+        internal Container<Symbol> SectionsParagraphs
```

Comment:
If the CFG builds its own Section and Paragraph symbol, maybe we could introduce an abstraction for both (`ProcedureSymbol`), this may simplify the code.

Reply:
See your post about adding a method in CfgSectionSymbol so that it can yield its sentences. I answered that CfgSectionSymbol must ihnerits from SectionSymbol and CfgParagraphSymbol inherits from ParagraphSymbol, so if you want to introduce this base class this shall be done in SDLite.

Reply:
Done in PR #1738.